pkg/arvo/sys/vane/clay.hoon

```diff
@@ -1962,32 +1968,163 @@
     =/  =wire  /merge/[syd]/(scot %p ali-ship)/[ali-desk]/[germ]
     (emit hen %pass wire %c %warp ali-ship ali-desk `[%sing %v case /])
   ::
+  ++  make-melt
+    |=  [bas=beak con=(list [beak germ])]
+    ^-  melt
+    :+  bas  con
+    %-  ~(gas by (~(put by *(map beak (unit dome:clay))) bas *(unit dome:clay)))
+    (turn con |=(a=[beak germ] [-.a *(unit dome:clay)]))
+  ::
+  ++  start-fuse
+    |=  [bas=beak con=(list [beak germ])]
+    ^+  ..start-fuse
+    =/  moves=(list move)
+        %+  turn
+          [[bas *germ] con]
+        |=  [bec=beak germ]
+        ^-  move
+        =/  wir=wire  /fuse/[syd]/(scot %p p.bec)/[q.bec]/(scot r.bec)
+        [hen %pass wir %c %warp p.bec q.bec `[%sing %v r.bec /]]
+    ::  we also want to clear the state (fiz) associated with this
+    ::  merge and print a warning if it's non trivial i.e. we're
+    ::  starting a new fuse before the previous one terminated.
+    ~&  ?~  con.fiz
+          [%starting-fuse bas con]
+        :*  %starting-fuse
+            bas
+            con
+            %discarding-state
+            ::  we don't want to ~& an entire dome
+            %-  ~(run by sto.fiz)
+            |=(v=(unit dome:clay) ?~(v %not-received %recieved))
+        ==
+    =.  fiz  (make-melt bas con)
+    (emil moves)
+  ::
+  ++  take-fuse
+    |=  [bec=beak =riot]
+    ^+  ..take-fuse
+    ?~  riot
+      ~&  [%fuse-for syd %missing bec]
+      ::  by setting fiz to *melt the merge is aborted - any further
+      ::  responses we get for the merge will cause take-fuse to crash
+      =.  fiz  *melt
+      ..take-fuse
+    ?>  (~(has by sto.fiz) bec)
+    =.  fiz
+        :+  bas.fiz  con.fiz
+        (~(put by sto.fiz) bec `!<(dome:clay q.r.u.riot))
+    =/  all-done=flag
+        %-  ~(all by sto.fiz)
+        |=  res=(unit dome:clay)
+        ^-  flag
+        !=(res ~)
+    ?.  all-done
+      ..take-fuse
+    =|  rag=rang
+    =/  clean-hut-ran  hut.ran
+    =/  initial-dome=dome:clay  (need (~(got by sto.fiz) bas.fiz))
+    =/  continuation-yaki=yaki
+        (~(got by hut.ran) (~(got by hit.initial-dome) let.initial-dome))
+    =/  parents=(list tako)  ~[(~(got by hit.initial-dome) let.initial-dome)]
+    =/  merges  con.fiz
+    |-
+    ^+  ..take-fuse
+    ?~  merges
+      ::  reset fiz before potentially parking - we hope to
+      ::  succeed so we don't need this state anymore.
+      =.  fiz  *melt
+      =.  hut.ran  clean-hut-ran
+      (park | [%| continuation-yaki(p (flop parents))] rag)
```

Comment:
Since we give `%mere` in the failure case, we should give it here as well.  I think it should come *after* the `park`, call (so the `%mere` arrives after any moves caused by the commit), so something like

```
=.  ..take-fuse  (park ...)
(done ...)
```

Reply:
Although I note that `|merge` sends the `%mere` before the `park`, so maybe that's fine too.